frontend/src/app/store/reducers/code-table.ts

```diff
@@ -797,100 +797,87 @@ export const selectComplaintStatusCodeDropdown = (state: RootState): Array<Optio
 };
 
 // This returns the complaint status codes, with the pending status.
-export const selectComplaintStatusWithPendingCodeDropdown = (state: RootState): Array<Option> => {
-  const {
-    codeTables: { "complaint-status": complaintStatus },
-  } = state;
-
-  const data = complaintStatus.map(({ complaintStatus, longDescription }) => {
-    const item: Option = { label: longDescription, value: complaintStatus };
-    return item;
-  });
-  return data;
-};
-
-export const selectSpeciesCodeDropdown = (state: RootState): Array<Option> => {
-  const {
-    codeTables: { species },
-  } = state;
-
-  const data = species.map(({ species, longDescription, isActive }) => {
-    const item: Option = { label: longDescription, value: species, isActive };
-    return item;
-  });
-  return data;
-};
-
-export const selectViolationCodeDropdown =
-  (agency: string) =>
-  (state: RootState): Array<Option> => {
-    const {
-      codeTables: { violation },
-    } = state;
-
-    const data = violation
-      .filter(({ agencyCode }) => agencyCode === agency)
-      .map(({ violation, longDescription, isActive }) => {
-        const item: Option = { label: longDescription, value: violation, isActive };
-        return item;
-      });
-    return data;
-  };
-
-export const selectGirTypeCodeDropdown = (state: RootState): Array<Option> => {
-  const {
-    codeTables: { "gir-type": girType },
-  } = state;
+export const selectComplaintStatusWithPendingCodeDropdown = createSelector(
```

Comment:
I refactored most of these to use createSelector as I noticed some additional infinite re-render scenarios related to the code table selectors.